### PR TITLE
[explorer] - deprecate DisplayBox

### PR DESCRIPTION
**Pull request by mamos-mysten**
This PR removes the `DisplayBox` component in favor of a simpler `ObjectDetails` component that removes some of the complexity by using `headless-ui` for handling the image preview. In addition: 

- added a `useImage` hook which handles the logic of formatting `ipfs://` URLs as well as image moderation
- remove `ImageModeratorClient` and replace with `useImageMod` hook that uses react-query
- adds a generic `Modal` / 'ImageModal' component for showing a full-screen preview
- updates the modded / fallback images to match designs

sample url: 
https://explorer-git-mamos-explorer-displaybox-mysten-labs.vercel.app/object/0x4deb7dc0f747ecc74f28e62ab7c8595d6420854452a75f19bdb42afe1d55de1c?network=testnet

### CODE REVIEW

File: apps/explorer/src/components/ownedobjects/OwnedObjectConstants.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/OwnedObjectConstants.tsx:16]: Curious why we're bumping

File: apps/explorer/src/ui/ObjectDetails.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/ObjectDetails.tsx:122]: If we're introducing a base image component can we make this extracted into it's own UI component? We could also improve the loading state and error behavior around this, e.g.: https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/image/src/use-image.ts

File: apps/explorer/src/ui/ObjectDetails.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/ObjectDetails.tsx:109]: This could be part of `Image`, right?

File: apps/explorer/src/ui/ObjectDetails.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/ObjectDetails.tsx:63]: Looks like we don't pass nsfw through here, is that intended?

File: apps/explorer/src/ui/ObjectDetails.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/ObjectDetails.tsx:77]: This is a behavior change I think, can you verify this is acceptable? Kevin probably has context here.

File: apps/explorer/src/ui/LightBox.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/LightBox.tsx:16]: Could we make a more generic `Modal` component?

File: apps/explorer/src/hooks/useMedia.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useMedia.ts:8]: Can you default nsfw to true for a small period of time to allow the image mod service time to respond, so that we don't flash potentially sensitive content on the page while we wait for a response?

File: apps/explorer/src/hooks/useImageMod.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImageMod.ts:5]: Thanks for rewriting this!

File: apps/explorer/src/hooks/useImageMod.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImageMod.ts:9]: Maybe make this take some variable time for better parity with prod?

File: apps/explorer/src/components/ownedobjects/views/OwnedNFTView.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/OwnedNFTView.tsx:28]: Key can just be the id

File: apps/explorer/src/components/ownedobjects/OwnedObjectConstants.tsx
Comment (mamos-mysten) [reply apps/explorer/src/components/ownedobjects/OwnedObjectConstants.tsx:16]: bumped it to 8 to align with the default in Figma: 

image

https://www.figma.com/file/kSD0qXCIDZA4KQVNepA5hR/02-Explorer?node-id=204-54037&t=qH5XGBaO26cfBdWs-4

File: apps/explorer/src/ui/ObjectDetails.tsx
Comment (mamos-mysten) [reply apps/explorer/src/ui/ObjectDetails.tsx:80]: verified with Kevin - updated UI works for sensitive images

**Comment by mamos-mysten** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8276** <a href="https://app.graphite.dev/github/pr/MystenLabs/sui/8276" target="_blank">Graphite</a>  👈
    * **PR #10864** <a href="https://app.graphite.dev/github/pr/MystenLabs/sui/10864" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/MystenLabs/sui/8276?utm_source=stack-comment).

File: apps/explorer/src/ui/image/Image.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/image/Image.tsx:47]: Doesn't seem to be used, probably missed passing this to the hook?

File: apps/explorer/src/ui/image/Image.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/image/Image.tsx:98]: Looks like moderate should be passed here?

File: apps/explorer/src/ui/image/Image.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/image/Image.tsx:75]: 🙈

File: apps/explorer/src/ui/image/Image.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/image/Image.tsx:78]: Is this the desired behavior? We can follow-up async after this merges but I think we currently fail open

File: apps/explorer/src/hooks/useImage.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImage.ts:38]: What if the moderation isn't done?

IMO I would make image load status and moderations status _fully_ decoupled, and just combine them in the hook return so that the result is identical but you have less intermediate logic here

File: apps/explorer/src/hooks/useImage.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImage.ts:39]: _maybe_ use `decode()`? IDK if that's ideal I didn't finish reading the MDN page: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decode

File: apps/explorer/src/hooks/useImageMod.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImageMod.ts:40]: Probably should be the same as the server, 24 hours.

File: apps/explorer/src/hooks/useImageMod.ts
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useImageMod.ts:27]: FWIW I hate nested await like this, much prefer just intermediate variables

**Review by Jordan-Mysten — APPROVED** [review]
Code LGTM, some small thoughts and questions but nothing blocking.

**Pull request merged**
